### PR TITLE
updated readme & addde code for thr task

### PR DESCRIPTION
**Pull request by dihyidi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,28 @@
 # lab-8-new
-Creating a simple Java maven project.
+Create a Java maven project.
```
Comment (PASSTor) [inline README.md:2]: Copying my comment from the Slack:
Итого, README должен:
Содержать условие вашей задачи
Содержать способ запуска вашей программы
Содержать описание алгоритма или принципа решения
Быть написан с помощью Markdown

File: README.md [hunk 2]
```diff
@@ -1,2 +1,28 @@
...
+1. Write the code for the task from the laboratory №2
```
Comment (PASSTor) [inline README.md:3]: So where's the task of your lab #2?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,28 @@
...
+2. When writing a program, you need to use the arrangements for the design of the java code convention.
+3. Classes need to be competently arranged in packages.
+4. Work with the console or the console menu should be minimal.
+5. Only the classes contained in the class diagram should be present in the code
```
Comment (PASSTor) [inline README.md:7]: So where's the diagram? Would you please care to add it to README?

File: code/pom.xml [hunk 4]
```diff
@@ -0,0 +1,148 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>ua.lviv.iot</groupId>
+  <artifactId>code</artifactId>
```
Comment (PASSTor) [inline code/pom.xml:8]: yeah... Do you think this is a good name for artifact?

File: code/pom.xml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  <version>1.0-SNAPSHOT</version>
+
+  <name>code</name>
+  <!-- FIXME change it to the project's website -->
```
Comment (PASSTor) [inline code/pom.xml:12]: Do you know what's FIXME for? ^_^

File: code/src/main/java/ua/lviv/iot/App.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package ua.lviv.iot;
```
Comment (PASSTor) [inline code/src/main/java/ua/lviv/iot/App.java:1]: This package name is good, but not good enough. Imaging someone will start using your app as a library (to import and reuse your models classes for example) – would it be clear what they deal with if they import ua.lviv.iot.* ? Not really, some more specifics would be better.

File: code/src/main/java/ua/lviv/iot/App.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                Material.CARBON_STEEL, Furniture.SOFA, 800);
+        jigsaw.function();
+
+        HandTool hacksaw = new HandTool("Hacksaw", "H236", "UaHack",
+                Material.LOW_TUNGSTEN_STEEL, Furniture.TABLE, 150);
+        hacksaw.function();
+
+        PowerTool grinder = new PowerTool("Grinder", "P789", "AGrinder",
+                Material.HIGH_SPEED_STEEL, Furniture.WARDROBE, 1000);
+        grinder.function();
```
Comment (PASSTor) [inline code/src/main/java/ua/lviv/iot/App.java:31]: function called function? :)

File: code/src/main/java/ua/lviv/iot/manager/WorkshopManager.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.stream.Collectors;
+
+import ua.lviv.iot.models.tools.Tool;
+import ua.lviv.iot.models.tools.PowerTool;
+import ua.lviv.iot.models.types.Furniture;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public final class WorkshopManager {
```
Comment (PASSTor) [inline code/src/main/java/ua/lviv/iot/manager/WorkshopManager.java:18]: Why do you choose to make this class final?

Comment (PASSTor) [reply code/src/main/java/ua/lviv/iot/manager/WorkshopManager.java:18]: P.S. This is a question to answer, not to fix

File: code/src/main/java/ua/lviv/iot/manager/WorkshopManager.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        return sortedByProducer;
+    }
+
+    public List<PowerTool> sortByPower(final SortOrder order) {
+        List<PowerTool> sortedByPower = new ArrayList<>();
+
+        for (Tool tool: listOfTools) {
+            if (tool instanceof PowerTool) {
```
Comment (PASSTor) [inline code/src/main/java/ua/lviv/iot/manager/WorkshopManager.java:51]: So you are saying if I call one `sort` method of this class I may get more elements back than when I call another `sort` method? Hmm...

File: code/src/main/java/ua/lviv/iot/models/tools/HandTool.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+@EqualsAndHashCode(callSuper = true)
+public final class HandTool extends Tool {
+    private int lengthMm;
+
+    @Override
+    public void function() {
+        System.out.println("Needs somebody to work");
+    }
+
+    public HandTool(final String name, final String model, final String producer,
```
Comment (PASSTor) [inline code/src/main/java/ua/lviv/iot/models/tools/HandTool.java:22]: So remind me what's the point of `@AllArgsConstructor`?

File: code/src/main/java/ua/lviv/iot/models/tools/HandTool.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+                    final Material material,
+                    final Furniture furnitureType,
+                    final int lengthMm) {
+        super(name, model, producer, material, furnitureType);
+        this.lengthMm = lengthMm;
+    }
+
+    @Override
+    public String toString() {
```
Comment (PASSTor) [inline code/src/main/java/ua/lviv/iot/models/tools/HandTool.java:31]: Doesn't `@Data` does that?

File: code/src/test/java/ua/lviv/iot/AppTest.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
+package ua.lviv.iot;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest
```
Comment (PASSTor) [inline code/src/test/java/ua/lviv/iot/AppTest.java:10]: Where have you copypasted this class from?

**Review by PASSTor — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dihyidi) [reply README.md:2]: fixed, thank you

File: README.md (hunk 2 above)
Comment (dihyidi) [reply README.md]: added, thank you

File: README.md (hunk 3 above)
Comment (dihyidi) [reply README.md]: added, thank you

File: code/pom.xml (hunk 4 above)
Comment (dihyidi) [reply code/pom.xml]: fixed, thank you

File: code/pom.xml (hunk 5 above)
Comment (dihyidi) [reply code/pom.xml]: fixed, thank you

**Pull request closed without merging**